crates/aranya-client/src/aqc/net.rs

```diff
@@ -0,0 +1,671 @@
+                    // This will update the channel map with the PSK and associate it with an
+                    // AqcChannel.
+                    if identity == PSK_IDENTITY_CTRL {
+                        if let ControlFlow::Break(_) =
+                            receive_ctrl_message(&daemon, &mut channel_map, &mut conn).await
+                        {
+                            continue;
+                        }
+                    // If the PSK identity hint is not the control PSK, check if it's in the channel map.
+                    // If it is, create a channel of the appropriate type. We should have already received
+                    // the control message for this PSK, if we don't we can't create a channel.
+                    } else if let Some(channel_info) = channel_map.get(identity) {
+                        tracing::debug!(
+                            "Found channel info in map for identity hint {:02x?}: {:?}",
+                            identity,
+                            channel_info
+                        );
+                        if let ControlFlow::Break(_) =
+                            create_channel_type(&sender, conn, channel_info).await
+                        {
+                            return;
+                        }
+                    } else {
+                        tracing::debug!(
+                            "No channel info found in map for identity hint {:02x?}",
+                            identity
+                        );
+                        continue;
+                    }
+                } else {
+                    tracing::warn!("No identity hint received. Unable to create channel.");
+                }
+            }
+            None => {
+                debug!("Server connection terminated");
+                break;
+            }
+        }
+    }
+}
+
+async fn create_channel_type(
+    sender: &mpsc::Sender<AqcChannelType>,
+    conn: Connection,
+    channel_info: &AqcChannel,
+) -> ControlFlow<()> {
+    match channel_info {
+        AqcChannel::Bidirectional { id } => {
+            // Once we accept a valid connection, let's turn it into an AQC Channel that we can
+            // then open an arbitrary number of streams on.
+            let (channel, bi_sender) =
+                AqcBidirectionalChannel::new(LabelId::default(), *id, conn.handle());
+
+            // Notify the AfcClient that we've accepted a new connection, which the user will
+            // have to call receive_channel() on in order to use.
+            if sender
+                .send(AqcChannelType::Bidirectional { channel })
+                .await
+                .is_ok()
+            {
+                // Spawn a new task so that we can receive any future streams that are opened
+                // over the connection.
+                tokio::spawn(handle_bidi_streams(conn, bi_sender));
```

Comment:
The API shouldn't be spawning tasks or using mpsc channels. E.g., AqcBidirectionalChannel should be something like

```rust
struct AqcBidirectionalChannel { ch: s2n_quic::Channel }
impl AqcBididirectionalChannel {
    async fn recv(&mut self) -> Result<Bytes> { ... }
    async fn send(&mut self, data: Bytes) -> Result<()>;
}
```

Reply:
> You can't send or recv on a channel. Only on streams.

Okay, then:

```rust
struct AqcBidirectionalChannel { ch: s2n_quic::Channel }
impl AqcBididirectionalChannel {
    async fn open_stream(&self) -> AqcBidiChannelStream { ... }
}
struct AqcBidiStream { ... }
impl AqcBidiStream {
    async fn recv(&mut self) -> Result<Bytes> { ... }
    async fn send(&mut self, data: Bytes) -> Result<()>;
    async fn open_stream(&self) -> AqcBidiChannelStream { ... }
}
```